File: android/src/main/java/com/ahmedadeltito/photoeditor/TranslationService.java
package com.ahmedadeltito.photoeditor;

import java.util.HashMap;

/**
 * Created by Thien PV on 11/24/23.
 */
public class TranslationService {

    private static TranslationService instance = null;
    private HashMap<String, String> translations;

    private TranslationService() {
        translations = new HashMap<>();
    }

    public static TranslationService getInstance() {
        if (instance == null) {
            instance = new TranslationService();
        }
        return instance;
    }

    public void init(HashMap translations) {
        this.translations = translations;
    }

    public String getTranslation(String key) {
        return translations.get(key);
    }
}

File: photo-editor/Photo Editor/Photo Editor/TranslationService.swift
@objc(TranslationService)
public final class TranslationService: NSObject {
    @objc public static let shared = TranslationService()

    private var translations: [String: String] = [:]

    private override init() {}

    @objc public func initTranslations(_ translations: [String: String]) {
        self.translations = translations
    }

    func getTranslation(for key: String) -> String? {
        let value = translations[key]
        return translations[key]
    }
}
